Assets/MixedRealityToolkit/_Core/Devices/BaseSpatialObserver.cs

```diff
@@ -0,0 +1,179 @@
+        /// </summary>
+        public bool IsRunning { get; protected set; }
+
+        /// <summary>
+        /// The collection of mesh <see cref="GameObject"/>s that have been observed.
+        /// </summary>
+        public virtual IDictionary<int, GameObject> Meshes => new Dictionary<int, GameObject>();
+
+        /// <summary>
+        /// Start the observer.
+        /// </summary>
+        public virtual void StartObserving()
+        { }
+
+        /// <summary>
+        /// Stop the observer.
+        /// </summary>
+        public virtual void StopObserving()
+        { }
+
+        /// <summary>
+        /// When a mesh is created we will need to create a game object with a minimum 
+        /// set of components to contain the mesh.  These are the required component types.
+        /// </summary>
+        private System.Type[] requiredMeshComponents =
+        {
+            typeof(MeshFilter),
+            typeof(MeshRenderer),
+            typeof(MeshCollider)
+        };
+
+        /// <summary>
+        /// The collection of meshes being managed by the observer.
+        /// </summary>
+        protected Dictionary<int, SpatialMeshObject> meshObjects = new Dictionary<int, SpatialMeshObject>();
+
+        /// <summary>
+        /// Creates a <see cref="SpatialMeshObject"/>.
+        /// </summary>
+        /// <param name="mesh"></param> todo: add comments
+        /// <param name="name"></param>
+        /// <param name="meshId"></param>
+        /// <returns>
+        /// SpatialMeshObject containing the fields that describe the mesh.
+        /// </returns>
+        protected SpatialMeshObject CreateSpatialMeshObject(
+            Mesh mesh,
+            string name,
+            int meshId)
+        {
+            SpatialMeshObject newMesh = new SpatialMeshObject();
+
+            newMesh.Id = meshId;
+            newMesh.GameObject = new GameObject(name, requiredMeshComponents);
+            newMesh.GameObject.layer = MixedRealityManager.SpatialAwarenessSystem.MeshPhysicsLayer;
+
+            newMesh.Filter = newMesh.GameObject.GetComponent<MeshFilter>();
+            newMesh.Filter.sharedMesh = mesh;
+
+            newMesh.Renderer = newMesh.GameObject.GetComponent<MeshRenderer>();
+
+            // Reset the surface mesh collider to fit the updated mesh. 
+            // Unity tribal knowledge indicates that to change the mesh assigned to a
+            // mesh collider, the mesh must first be set to null.  Presumably there
+            // is a side effect in the setter when setting the shared mesh to null.
+            newMesh.Collider = newMesh.GameObject.GetComponent<MeshCollider>();
+            newMesh.Collider.sharedMesh = null;
+            newMesh.Collider.sharedMesh = newMesh.Filter.sharedMesh;
+
+            return newMesh;
+        }
+
+        /// <summary>
+        /// Cleans up mesh objects managed by the observer.
+        /// </summary>
+        protected void CleanupMeshes()
```

Comment:
Shouldn't this also be part of the interface?

Reply:
nope, part of the internal observer details. no one calls them from outside. this is to be called from Destroy() on the observer